### PR TITLE
[BottomAppBar Demo] Ensured fab is traversed before bottomAppBar buttons by accessibility…

### PR DESCRIPTION
**Pull request by harperl-lgtm**
This PR made sure that in BottomAppBar demo the FAB is traversed before BottomAppBar buttons by accessibility focus, which is more intuitive considering that FAB is always above the BottomAppBar buttons.

Issues fixed:
[b/238426806]

https://user-images.githubusercontent.com/81855102/207170870-b0b454a5-08c5-4249-8846-ec215ed433b4.mp4

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: lib/demos/material/bottom_app_bar_demo.dart [hunk 1]
```diff
@@ -165,21 +170,30 @@ class _DemoBottomAppBar extends StatelessWidget {
         data: IconThemeData(color: Theme.of(context).colorScheme.onPrimary),
         child: Row(
           children: [
-            IconButton(
-              tooltip: MaterialLocalizations.of(context).openAppDrawerTooltip,
-              icon: const Icon(Icons.menu),
-              onPressed: () {},
+            Semantics(
+              sortKey: const OrdinalSortKey(1),
```
Comment (guidezpl) [inline lib/demos/material/bottom_app_bar_demo.dart:174]: I'm curious if this works without 1, 2, and 3. For example, if you only add it above the row

```
Semantics(
 sortKey: const OrdinalSortKey(1),
 Row(
...
```

File: pubspec.lock [hunk 2]
```diff
@@ -5,140 +5,160 @@ packages:
     dependency: transitive
     description:
       name: _fe_analyzer_shared
-      url: "https://pub.dartlang.org"
+      sha256: "3444216bfd127af50bbe4862d8843ed44db946dd933554f0d7285e89f10e28ac"
```
Comment (guidezpl) [inline pubspec.lock:8]: please revert

**Review by guidezpl — CHANGES_REQUESTED** [review]

File: lib/demos/material/bottom_app_bar_demo.dart (hunk 1 above)
Comment (harperl-lgtm) [reply lib/demos/material/bottom_app_bar_demo.dart:174]: The Row currently does not have its own SemanticNode, which makes its 3 buttons siblings to the fab. In order for this to work we need to add a SemanticNode (and focus) for the Row itself. It's just the focus might not look great when the fab is docked and the BottomAppBar has a notch.

Comment (guidezpl) [reply lib/demos/material/bottom_app_bar_demo.dart:174]: If you wrap the row in Semantics, can you avoid having three separate Semantics for its button children?

Comment (harperl-lgtm) [reply lib/demos/material/bottom_app_bar_demo.dart:174]: Yes I think so. Lmk if you are leaning towards this

Comment (guidezpl) [reply lib/demos/material/bottom_app_bar_demo.dart:174]: If we can achieve this with a simpler solution (2 semantic nodes), let's go with that

Comment (harperl-lgtm) [reply lib/demos/material/bottom_app_bar_demo.dart]: Sounds good! Updated.

**Comment by esouthren** [comment]
LGTM - but can we test this in some way? A verification screencast of the accessibility focus now working as intended would be great.

**Pull request merged**